examples/platform/policy_data_collection.py

```diff
@@ -0,0 +1,132 @@
+This is an example of the command line to execute.
+> python examples/platform/policy_data_collection.py --profile EXAMPLE_CREDENTIALS list_policies
+
+The second method enables Auth Event collection (a type of data collection rule) on an existing policy.
+Command line prompts ask for required info and defaults are provided.
+This is an example of the command line to execute. The script will prompt on the command line for Policy Id.
+> python examples/platform/policy_data_collection.py --profile EXAMPLE_CREDENTIALS enable_auth_event_rule
+
+To disable an Auth Event policy,
+> python examples/platform/policy_data_collection.py --profile EXAMPLE_CREDENTIALS disable_auth_event_rule
+
+The API specification for underlying APIs is available on the Developer Network:
+https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/policy-service/#rule-config---data-collection
+"""
+
+import sys
+from cbc_sdk.helpers import build_cli_parser, get_cb_cloud_object
+from cbc_sdk.platform import Policy
+
+
+def list_policy_summaries(cb, args):
+    """List all policies and their rules."""
+    # the cb.select(Policy) with no parameters will get all the policies for the organization
+    for p in cb.select(Policy):
+        print(u"Policy id {0}: {1} {2}".format(p.id, p.name, "({0})".format(p.description) if p.description else ""))
+        print("Data Collection Rules:")
+        if p.data_collection_rule_configs is None:
+            print("No Data Collection Rules")
+        elif len(p.data_collection_rule_configs_list) == 0:
+            print("No Data Collection Rules")
+        else:
+            for dcrc in p.data_collection_rule_configs_list:
+                print(dcrc)
+                print("the schema for this data collection")
+                print(p.get_ruleconfig_parameter_schema(dcrc.id))
+
+        print("")
+        print("End of Policy Object")
+        print("")
+
+
+def enable_auth_event_rule(cb):
+    """Enable the rule to collect Auth Events from the sensor"""
+    # prompt the user for a policy Id
+    user_input = input("Enter the policy Id on which to enable collection")
+    policy = cb.select(Policy, user_input)
+    print("Using policy id: {0}.  name: {1}".format(policy.id, policy.name))
+    # If the policy does not have a data collection section then fail
+    if len(policy.data_collection_rule_configs_list) == 0:
+        print("No data collection elements available to enable")
+        exit()
+
+    auth_event_rule_config = None
+    for dcrc in policy.data_collection_rule_configs_list:
+        if dcrc.name == 'Authentication Events':
+            dcrc.set_parameter('enable_auth_events', True)
+            dcrc.save()
+            auth_event_rule_config = dcrc
+
+    if auth_event_rule_config is not None:
+        print(auth_event_rule_config)
+    else:
+        print("Auth Event Rule Config Element Not Found")
+
+
+def disable_auth_event_rule(cb):
```

Comment:
I wonder whether the enable / disable functions can be merged and just one function to be used as they share most of the code, but even that seems ok.

Reply:
agree.  merged.